lib/merge.ts

```diff
@@ -7,7 +7,7 @@ export const mergeCustomizer = (
   key: string,
   object: any,
 ) => {
-  if (Array.isArray(srcVal)) {
+  if (Array.isArray(srcVal) || ArrayBuffer.isView(srcVal)) {
     return srcVal;
```

Comment:
Great solution. To help with future maintenance, I'd love to add a comment here

```suggestion
  if (Array.isArray(srcVal) || ArrayBuffer.isView(srcVal)) {
    // don't try to merge arrays or subclasses of TypedArray
    // ArrayBuffer.isView() matches Buffer, Uint8Array, Float32Array, etc.
    return srcVal;
```